### PR TITLE
ANNULUS-293 Cached Networks

### PR DESCRIPTION
**Pull request by berriose**
## Description

- pull networks from cache
- add a new network to cache

Fixes # (issue)

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Manual

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] I have added screenshots

### CODE REVIEW

File: lib/chain/models/currency.dart [hunk 1]
```diff
@@ -15,3 +15,7 @@ List<String> currencies = [
   'ZAR',
   'AUD',
 ];
+
+Currency convertStringToCurrency({required String currencyString}) {
```
Comment (KalervoHyyppa) [inline lib/chain/models/currency.dart:19]: You can delete the list of strings and just used an enhance `enum` and add `string` as a property. See chart_option.dart for an example

File: lib/chain/providers/selected_chain_provider.dart [hunk 2]
```diff
@@ -1,7 +1,102 @@
...
+      const Chains.private_network(),
+      const Chains.dev_network(),
+      const Chains.mock(),
+    ];
+
+    //state holds both standard and custom chains
+    final allChains = [...standardChains, ...customChains];
+
+    if (setState) {
+      Future.delayed(const Duration(seconds: 1), () {
```
Comment (KalervoHyyppa) [inline lib/chain/providers/selected_chain_provider.dart:52]: Remove `Future.delayed` it was here just to show loading UI

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
See comments

File: pubspec.yaml [hunk 3]
```diff
@@ -57,6 +57,8 @@ dependencies:
   intl: ^0.17.0
   super_tooltip: ^2.0.3
   fast_base58: ^0.2.1
+  hive: ^2.2.3
+  hive_flutter: ^1.1.0
```
[the remark this answers is not in the document]
Comment (berriose) [reply pubspec.yaml:61]: Yes, we need hive for the hive cache and hive_flutter to use it with flutter apps.

File: lib/chain/models/chains.dart [hunk 4]
```diff
@@ -1,37 +1,82 @@
...
+List<dynamic> splitUrl({required String completeUrl}) {
+  // '<route:port>'
+  final splitUrl = completeUrl.split(':');
+
+  if (splitUrl.length > 2) {
+    throw Exception('Invalid Url');
+  }
+
+  var port = int.tryParse(splitUrl[1]);
+  if (int == null) {
```
Comment (KalervoHyyppa) [inline lib/chain/models/chains.dart:57]: I think you meant to use `port` here

File: lib/chain/sections/chainname_dropdown.dart
Comment (KalervoHyyppa) [inline lib/chain/sections/chainname_dropdown.dart:1]: When adding a new chain, it doesn't automatically appear in the dropdown

**Review by KalervoHyyppa — CHANGES_REQUESTED** [review]
A couple of more changes

File: lib/chain/models/chains.dart (hunk 4 above)
Comment (berriose) [reply lib/chain/models/chains.dart:57]: thanks for catching this.

**Review by KalervoHyyppa — APPROVED** [review]
Nice, it's good to go

**Pull request merged**